File: lib/models/role.dart
import 'employee.dart';

class Role {
  final int id;
  final String name;
  final List<Employee> employees;

  Role({
    required this.id,
    required this.name,
    required this.employees,
  });

  factory Role.fromJson(Map<String, dynamic> json) {
    return Role(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      employees: (json['employees'] as List<dynamic>? ?? [])
          .map((emp) => Employee.fromJson(emp))
          .toList(),
    );
  }
}

File: app/Models/Role.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;

class Role extends Model
{
    use HasFactory;

    protected $fillable = [
        'business_id',
        'name',
    ];

    public function business(): BelongsTo
    {
        return $this->belongsTo(Business::class);
    }

    public function employees(): BelongsToMany
    {
        return $this->belongsToMany(Employee::class, 'employee_roles')
            ->withPivot('business_id', 'assigned_at')
            ->withTimestamps();
    }

    public function permissions(): BelongsToMany
    {
        return $this->belongsToMany(Permission::class, 'role_permissions')
            ->withPivot('granted_at')
            ->withTimestamps();
    }
}
